docs/book/tutorials/kubernetes-on-vsphere-with-kubeadm.md

```diff
@@ -156,262 +208,994 @@ bootstrapTokens:
+kube-scheduler-k8s-master            1/1       Running   0          37s
 ```
-
-Then upload it to the worker nodes.
-
-```sh
-scp discovery.yaml ubuntu@10.198.17.189:~/discovery.yaml
-scp discovery.yaml ubuntu@10.198.17.190:~/discovery.yaml
-scp discovery.yaml ubuntu@10.198.17.191:~/discovery.yaml
 ```
+# kubectl describe pod coredns-fb8b8dccf-q57f9 --namespace=kube-system
+.
+.
+Events:
+  Type     Reason            Age                 From               Message
+  ----     ------            ----                ----               -------
+  Warning  FailedScheduling  7s (x21 over 2m1s)  default-scheduler  0/1 nodes are available: 1 node(s) had taints that the pod didn't tolerate.
+```
+### Install flannel pod overlay networking
+The next step that needs to be carried out on the master node is that the flannel pod overlay network must be installed so the pods can communicate with each other. 
+The command to install flannel on the master is as follows:
+```
+# kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/62e44c867a2846fefb68bd5f178daf4da3095ccb/Documentation/kube-flannel.yml
+```
+Please follow these alternative instructions to install a pod overlay network other than flannel.
 
-#### On the worker nodes
-
-To check and make sure the `discovery.yaml` file was copied correctly, do a quick `cat`.
 
-```sh
-cat ~/discovery.yaml
+### Export the master node configuration
+Finally, the master node configuration needs to be exported as it is used by the worker nodes wishing to join to the master.
 ```
+# kubectl -n kube-public get configmap cluster-info -o jsonpath='{.data.kubeconfig}' > discovery.yaml
+```
+The discovery.yaml file will need to be copied to each of the worker nodes at /etc/kubernetes/discovery.yaml
 
-Then create the worker node `kubeadm` config yaml file (notice it's using our `discovery.yaml` as the input for master discovery) and the `token` is the same as we put in the master `kubeadminitmaster.yaml` configuration above and we specify the `cloud-provider` as `vsphere` for the workers:
-
-```yaml
-sudo tee /etc/kubernetes/kubeadminitworker.yaml >/dev/null <<EOF
+## Installing the Kubernetes worker node(s)
+Perform this task on the worker nodes.
+To have the worker node(s) join to the master, a worker node kubeadm config yaml file must be created. Notice it is using /etc/kubernetes/discovery.yaml as the input for master discovery. We will show how to copy the file from the workers to the master in the next step. Also, notice that the token used in the worker node config is the same as we put in the master kubeadminitmaster.yaml configuration above. Finally, we once more specify that the cloud-provider is external for the workers, as we are going to use the new CPI:
+```
+# tee /etc/kubernetes/kubeadminitworker.yaml >/dev/null <<EOF
 apiVersion: kubeadm.k8s.io/v1beta1
 caCertPath: /etc/kubernetes/pki/ca.crt
 discovery:
   file:
-    kubeConfigPath: discovery.yaml
+    kubeConfigPath: /etc/kubernetes/discovery.yaml
```

Comment:
This won't work with the below `scp` commands - you're SCPing to `/home/ubuntu/discovery.yaml` - advise updating either the `scp` command or this to be explicit and exactly the same.

Reply:
Good catch - updated text to reflect what is needed